mod-source-record-manager-server/src/main/java/org/folio/dao/JobExecutionProgressDaoImpl.java

```diff
@@ -3,108 +3,112 @@
+import static org.folio.rest.persist.PostgresClient.convertToPsqlStandard;
 
 @Repository
 public class JobExecutionProgressDaoImpl implements JobExecutionProgressDao {
 
   private static final Logger LOGGER = LogManager.getLogger();
 
+  public static final String JOB_EXECUTION_ID_FIELD = "job_execution_id";
+  public static final String TOTAL_RECORDS_COUNT = "total_records_count";
+  public static final String PROCESSED_RECORDS_COUNT = "processed_records_count";
+  public static final String ERROR_RECORDS_COUNT = "error_records_count";
+
   private static final String TABLE_NAME = "job_execution_progress";
-  public static final String JOB_EXECUTION_ID_FIELD = "jobExecutionId";
+  private static final String INSERT_SQL = "INSERT INTO %s.%s (job_execution_id, total_records_count, processed_records_count, error_records_count) VALUES ($1, $2, $3, $4)";
+  private static final String SELECT_BY_JOB_EXECUTION_ID_QUERY = "SELECT * FROM %s.%s WHERE job_execution_id = $1";
+  private static final String UPDATE_SQL = "UPDATE %s.%s SET job_execution_id = $1, total_records_count = $2, processed_records_count = $3, error_records_count = $4 WHERE job_execution_id = $1";
 
   @Autowired
   private PostgresClientFactory pgClientFactory;
 
   @Override
   public Future<Optional<JobExecutionProgress>> getByJobExecutionId(String jobExecutionId, String tenantId) {
-    Promise<Results<JobExecutionProgress>> promise = Promise.promise();
-    Criteria jobIdCrit = constructCriteria(JOB_EXECUTION_ID_FIELD, jobExecutionId).setJSONB(false);
-    pgClientFactory.createInstance(tenantId).get(TABLE_NAME, JobExecutionProgress.class, new Criterion(jobIdCrit), true, false, promise);
-    return promise.future()
-      .map(Results::getResults)
-      .map(progressList -> progressList.isEmpty() ? Optional.empty() : Optional.of(progressList.get(0)));
+    Promise<RowSet<Row>> promise = Promise.promise();
+    String query = String.format(SELECT_BY_JOB_EXECUTION_ID_QUERY, convertToPsqlStandard(tenantId), TABLE_NAME);
+    Tuple queryParams = Tuple.of(jobExecutionId);
+    pgClientFactory.createInstance(tenantId).select(query, queryParams, promise);
+    return promise.future().map(this::mapResultSetToOptionalJobExecutionProgress);
   }
 
   @Override
   public Future<JobExecutionProgress> initializeJobExecutionProgress(String jobExecutionId, Integer totalRecords, String tenantId) {
     Promise<JobExecutionProgress> promise = Promise.promise();
     Promise<SQLConnection> tx = Promise.promise();
-    Criteria jobIdCrit = constructCriteria(JOB_EXECUTION_ID_FIELD, jobExecutionId).setJSONB(false);
-    PostgresClient client = pgClientFactory.createInstance(tenantId);
+    PostgresClient pgClient = pgClientFactory.createInstance(tenantId);
     Future.succeededFuture()
       .compose(v -> {
-        client.startTx(tx);
+        pgClient.startTx(tx);
         return tx.future();
       })
       .compose(sqlConnection -> {
```

Comment:
Can we combine this block (line from 64 to 82) with block 127 - 145) ?

Reply:
Thank you @SerhiiNosko , done with this, you could take a look.